uber/custom_tags.py

```diff
@@ -116,242 +176,67 @@ def email_only(email):
-        return '\n'.join(results)
-
-
-@tag
-class radio(template.Node):
-    def __init__(self, name, value, default):
-        self.name    = name[1:-1]
-        self.value   = Variable(value)
-        self.default = Variable(default)
-
-    def render(self, context):
-        value   = self.value.resolve(context)
-        default = self.default.resolve(context)
-        checked = 'checked' if str(value) == str(default) else ''
-        return """<div class="radio"><label class="btn btn-primary"><input type="radio" name="%s" value="%s" %s /></label></div>""" % (self.name, value, checked)
-
-
-@tag
-class radiogroup(template.Node):
-    def __init__(self, opts, field):
-        model, self.field_name = field.rsplit('.', 1)
-        self.model = Variable(model)
-        self.opts = Variable(opts)
-
-    def render(self, context):
-        model = self.model.resolve(context)
-        options = self.opts.resolve(context)
-        default = getattr(model, self.field_name, None)
-        results = []
-        for num, desc in options:
-            checked = 'checked' if num == default else ''
-            results.append('<label class="btn btn-default" style="text-align: left;"><input type="radio" name="{}" autocomplete="off" value="{}" onchange="donationChanged();" {} /> {}</label>'
-                           .format(self.field_name, num, checked, desc))
-        return ''.join(results)
-
-
-@tag
-class hour_day(template.Node):
-    def __init__(self, dt):
-        self.dt = Variable(dt)
-
-    def render(self, context):
-        return hour_day_format(self.dt.resolve(context))
-
-
-@tag
-class timespan(template.Node):
-    def __init__(self, model):
-        self.model = Variable(model)
-
-    @staticmethod
-    def pretty(model, minute_increment=60):
-        minutestr = lambda dt: ':30' if dt.minute == 30 else ''
-        endtime   = model.start_time_local + timedelta(minutes=minute_increment * model.duration)
-        startstr  = model.start_time_local.strftime('%I').lstrip('0') + minutestr(model.start_time_local)
-        endstr    = endtime.strftime('%I').lstrip('0') + minutestr(endtime) + endtime.strftime('%p').lower()
-
-        if model.start_time_local.day == endtime.day:
-            endstr += endtime.strftime(' %A')
-            if model.start_time_local.hour < 12 and endtime.hour >= 12:
-                return startstr + 'am - ' + endstr
-            else:
-                return startstr + '-' + endstr
+            selected = 'selected="selected"' if str(val) == str(default) else ''
+        val  = str(val).replace('"',  '&quot;').replace('\n', '')
+        desc = str(desc).replace('"', '&quot;').replace('\n', '')
+        results.append('<option value="{}" {}>{}</option>'.format(val, selected, desc))
+    return '\n'.join(results)
+
+
+@JinjaEnv.jinja_export()
+def int_options(minval, maxval, default="1"):
+    results = []
+    for i in range(minval, maxval+1):
+        selected = 'selected="selected"' if i == default else ''
```

Comment:
This line isn't going to work as-is.  One of the differences between Django and Jinja is that in Django custom tags, everything comes in as a string.  So if you say

```
{% foo 1 10 %}
```

Then you'll end up with ``'1'`` and ``'10'`` as strings and need to coerce them to integers.  Whereas in Jinja if you say

```
{{ foo(1, 10) }}
```

then ``1`` and ``10`` will come through as integers.

All of this is why we set ``default`` to ``"1"`` instead of ``1``, if memory serves.  However, the way the code is working now, you're using ``==`` to compare an integer ``i`` to the string ``default`` which means that they'll never be equal.

Fixing this should be as simple as changing ``default="1"`` to ``default=1``.